launch_socket_server 2.0.0

mistydemeo opened the pull request:
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

This restores `launch_socket_server`, which was deleted in #47838. In that PR, it was deleted because it was believed to be broken due to its test failing; in actuality, it turned out that its `brew test` block was buggy, but that the program itself works.

It turns out that our current sandbox implementation prevents the program from either loading new launch daemons or from binding to these ports; it's not quite clear which. The result is that the test, which once worked, now fails. We don't really have a way to fix the test, so I've gone for an alternate implementation similar to other server-based software we run. See some discussion here: https://github.com/Homebrew/homebrew-core/pull/47510#issuecomment-565776221

If there's concern over the source repo being archived, I'm happy to take over its maintenance burden. My company uses this software heavily so I'm very motivated to keep it working!

cc @iwittkau, @chenrui333

SMillerDev commented:
Should probably remove the revision and the requirement for Yosemite since the latter is already a homebrew requirement

iwittkau commented:
Hi @mistydemeo,

great job, fixing the test!

> If there's concern over the source repo being archived, I'm happy to take over its maintenance burden.

Yes, please! I think there should be a maintained repository of this software. An archived repository has some implications:

> When a repository is archived, its issues, pull requests, code, labels, milestones, projects, wiki, releases, commits, tags, branches, reactions, and comments become read-only. To make changes in an archived repository, you must unarchive the repository first.

https://help.github.com/en/github/creating-cloning-and-archiving-repositories/about-archiving-repositories

If there is no way of submitting upstream issues or PRs to the upstream of this formula there is no way of resolving problems in the future.

mistydemeo commented:
> Should probably remove the revision and the requirement for Yosemite since the latter is already a homebrew requirement

mistydemeo commented:
> Should probably remove the revision and the requirement for Yosemite since the latter is already a homebrew requirement

As long as the version is still at 1.0.0, revision should stay the same; otherwise, it'll be treated as older for people who still had it installed from pre-removal.

I've switched over to my fork and bumped to version 2.0.0. I removed the Yosemite dep and reset the revision.

mistydemeo commented:
Since this is a new maintainer's fork of an existing repo, those warnings are reasonably ignorable in this circumstance.

SMillerDev commented:
Yeah, makes sense. Why did you abort the build?

mistydemeo commented:
> Why did you abort the build?

The Catalina build failed due to failure to fetch from the repo. Happened again, however. https://jenkins.brew.sh/job/Homebrew%20Core%20Pull%20Requests/53541/version=catalina/console

SMillerDev commented:
Node disk seems to be full. I'll see in slack if someone has time to fix it.

mistydemeo commented:
@BrewTestBot build this please
